pyanaconda/modules/boss/installation.py

```diff
@@ -0,0 +1,151 @@
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+import os.path
+import shutil
+import glob
+
+from pyanaconda.anaconda_loggers import get_module_logger
+from pyanaconda.core import util
+from pyanaconda.flags import flags
+from pyanaconda.modules.common.task import Task
+
+
+log = get_module_logger(__name__)
+
+ANACONDA_LOG_DIR = "/var/log/anaconda/"
+
+
+class CopyLogsTask(Task):
+    """Task to copy logs to target system."""
+    def __init__(self, sysroot):
+        """Create a new task.
+
+        :param sysroot: a path to the root of installed system
+        :type sysroot: str
+        """
+        super().__init__()
+        self._sysroot = sysroot
+
+    @property
+    def name(self):
+        return "Copy installation logs"
+
+    def run(self):
+        self._copy_logs()
+        self._copy_kickstart()
+        self._relabel_files()
+
+    def _copy_logs(self):
+        """Copy installation logs to the target system"""
+        log.info("Copying logs from the installation environment...")
+
+        if flags.nosave_logs:
+            log.warning("Installation logs will not be saved to the installed system due to the "
+                        "nosave option.")
+        else:
+            self._create_logs_directory()
+            self._copy_tmp_logs()
+            self._copy_pre_script_logs()
+            self._copy_dnf_debugdata()
+            self._copy_post_script_logs()
+            self._dump_journal()
+            self._chmod_logs()
+
+            log.info("Done.")
+
+    def _create_logs_directory(self):
+        """Create directory for Anaconda logs on the install target"""
+        util.mkdirChain(self._sysroot + ANACONDA_LOG_DIR)
+
+    def _copy_tmp_logs(self):
+        """Copy a number of log files from tmp"""
+        LOG_FILES_TO_COPY = [
+            "anaconda.log", "syslog", "X.log", "program.log", "packaging.log", "storage.log",
+            "ifcfg.log", "lvm.log", "dnf.librepo.log", "hawkey.log", "dbus.log"
+        ]
+        for logfile in LOG_FILES_TO_COPY:
+            self._copy_file_to_sysroot("/tmp/" + logfile,
+                                       ANACONDA_LOG_DIR + logfile)
+        # copy also list of packages used for creating the installation media
```

Comment:
To be pedantic this does not seem to belong to the function according to its name, but feel free to keep it here.